### PR TITLE
Add pinning retry worker

### PR DESCRIPTION
**Pull request by en0ma**
This PR replaces the existing `refreshPinQueue` implementation with a pinning monitor/retry worker. This new approach will not only refresh the pin queue when Estuary is restarted, but it will also handle missing pin requests while it runs.

**How it works**

- It runs every X hour interval (default is 1)
- It selects contents that have been created over 1hr that are still in the `pinning` state. So as not to stress the database; it batches the selection of the contents (default is 1000) to retry 

**Verifications**

- I have verified it picks up only contents that are an hour old that are still in `pinning` state
- I have verified it batches the selections correctly (only picks 1000 per loop until no more contents)
- I have verified contents are picked up and properly queued for re-pinning

### CODE REVIEW

**Review by ohmpatel1997 — APPROVED** [review]
looks good, thanks for taking this up.

File: main.go [hunk 1]
```diff
@@ -714,6 +714,7 @@ func main() {
 
 		go cm.Run(cctx.Context)                                                 // deal making and deal reconciliation
 		go cm.handleShuttleMessages(cctx.Context, cfg.RPCMessage.QueueHandlers) // register workers/handlers to process shuttle rpc messages from a channel(queue)
+		go cm.RunPinningRetryWorker(cctx.Context)                               // pinning retry worker, re-attempt pinning contents, not yet pinned after a period of time
```
Comment (alvin-reyes) [inline main.go:717]: Neat that you separated this.

File: pinning_retry_worker.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	timer := time.NewTicker(time.Hour * cm.cfg.Pinning.RetryWorker.Interval)
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-timer.C:
+			startContentID := 0
+			for {
+				var contents []util.Content
+				if err := cm.DB.Limit(cm.cfg.Pinning.RetryWorker.BatchLimit).Order("id ASC").Find(&contents, "pinning and not active and not failed and not aggregate and id > ? and created_at > NOW() + INTERVAL '1 HOURS'", startContentID).Error; err != nil {
```
Comment (alvin-reyes) [inline pinning_retry_worker.go:21]: I'm thinking we should add a date here. We should just re-try the last 2 to 3 months of pins. We will need to run another GC routine to clean up old data.

Comment (en0ma) [reply pinning_retry_worker.go:21]: ok

File: config/estuary.go [hunk 3]
```diff
@@ -101,6 +102,13 @@ func NewEstuary(appVersion string) *Estuary {
 			AggregateInterval:       time.Minute * 5,                                         // aggregate staging buckets every 5 minutes
 		},
 
+		Pinning: Pinning{
+			RetryWorker: RetryWorker{
+				Interval:   1,
```
Comment (gmelodie) [inline config/estuary.go:107]: This should be of type `time.Duration` and set to `1*time.Hour` (more descriptive, tells us this is in hours)

**Review by gmelodie — DISMISSED** [review]
Nice!

File: pinning_retry_worker.go (hunk 2 above)
Comment (jcace) [reply pinning_retry_worker.go:21]: `and created_at > NOW() + INTERVAL '1 HOURS'` 

Why is this hard coded to `1 HOURS` ? Wouldn't it need to be dynamic based on `cm.cfg.Pinning.RetryWorker.Interval` ?

Comment (en0ma) [reply pinning_retry_worker.go:21]: `cm.cfg.Pinning.RetryWorker.Interval` is how often it should check,, `and created_at > NOW() + INTERVAL '1 HOURS'` is what data it should select at the interval. I will make this dynamic with a default of 2 months back

**Review by alvin-reyes — APPROVED** [review]
LGTM. 

I think we should have a separate configuration constants for this limits. I'm thinking we need to have a separate GC process to clean up the data that we don't process anymore.

**Comment by en0ma** [comment]
> LGTM.
> 
> I think we should have a separate configuration constants for this limits. I'm thinking we need to have a separate GC process to clean up the data that we don't process anymore.

we can have another worker for GC. what limits are you talking about?

**Comment by ohmpatel1997** [comment]
@en0ma  @alvin-reyes  can we please migrate the existing stuck pins from last may be 6 months instead of 60 days?

**Comment by en0ma** [comment]
@alvin-reyes I think 6months can work too, let's do this

**Comment by ohmpatel1997** [comment]
> @alvin-reyes I think 6months can work too, let's do this

thanks

**Comment by ohmpatel1997** [comment]
@en0ma @alvin-reyes  just wanted to check if the above is deployed or not?

**Comment by ohmpatel1997** [comment]
Ohk, thanks

**Pull request merged**